### PR TITLE
draft: Video.js 7 roadmap

### PR DESCRIPTION
**Pull request by gkatsev**

### CODE REVIEW

**Comment by gkatsev** [comment]
Deploy preview for *videojs-blog* ready!

Built with commit b37b78ff605107bf64e19990bb106f4046c7ff6c

https://deploy-preview-27--videojs-blog.netlify.com

File: source/_drafts/Video-js-7-Roadmap.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+VHS will ship with Video.js by default because one of the guiding principles of Video.js is to make it easy for users to just put it on the page and have a player that works across browsers. With HLS and DASH becoming so popular, we thought it was about time for a plug and playable Video.js for the most common streaming formats.
+In addition to having it be included by default, we’ll make sure to provide builds that exclude VHS for those that don’t need it or are using another playback engine like HLS.js.
+
+## Old IE
+
+![IE8](IE8 small.png)
+
+Video.js has a long history of supporting IE. It has a flash fallback that was originally created for IE8. When Video.js 5 was released, we had a lot of code in place to support IE8 and when Video.js 6 came around, with Flash on its way out, we moved Flash support to a separate project, videojs-flash. Now, based on usage data we have, we are planning to remove support for IE8, IE9, and IE10.
```
Comment (mmcc) [inline source/_drafts/Video-js-7-Roadmap.md]: I might reword this first sentence a little. How do you feel about:

> Video.js has a long history of making a best effort to support IE, starting with a Flash fallback originally created for IE8.

The last sentence feels a little ominous. What about just:

> Now, based on usage data, we are planning ...

File: source/_drafts/Video-js-7-Roadmap.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+Today, I’m excited to announce the roadmap to Video.js 7! While this is a major version update, the actual changes are not that major. The two main changes are the addition of videojs-http-streaming, VHS for short, and the removal of support for older versions of Internet Explorer.
+
+![VJS 7](vjs 7 small.png)
+
+## Video.js HTTP Streaming
+
+![](VHS small.png)
+
+VHS used to be videojs-contrib-hls, but as MPEG-DASH became more popular, we realized that we wanted to support it as well and that we can share a lot of the code between HLS and DASH implementations. Look for a more details in an announcement post for VHS soon!
```
Comment (kcj382) [inline source/_drafts/Video-js-7-Roadmap.md]: Remove the 'a' from "Look for a more details..."?

File: source/_drafts/Video-js-7-Roadmap.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+In addition to having it be included by default, we’ll make sure to provide builds that exclude VHS for those that don’t need it or are using another playback engine like HLS.js.
+
+## Old IE
+
+![IE8](IE8 small.png)
+
+Video.js has a long history of supporting IE. It has a flash fallback that was originally created for IE8. When Video.js 5 was released, we had a lot of code in place to support IE8 and when Video.js 6 came around, with Flash on its way out, we moved Flash support to a separate project, videojs-flash. Now, based on usage data we have, we are planning to remove support for IE8, IE9, and IE10.
+According to the data we’ve gathered, IE in total has a share of about 4%. Of those 4%, IE8, IE9, and IE10 combined make up barely 5%, and IE11 has around 91% usage. This means that combined IE8, IE9, and IE10 are barely 0.002% of the total usage of Video.js. Based on this incredibly small usage, it is just not worth maintaining support for these browsers. Also, Video.js 6 isn’t going away anytime soon and will be available for your old IE support needs.
```
Comment (mmcc) [inline source/_drafts/Video-js-7-Roadmap.md]: How do you feel about:

> Based on this incredibly small ~usage~ footprint, we believe it isn't worth the the substantial effort required to maintain support for these outdated browsers. ~Also~ The good news is, for those concerned, Video.js 6 isn't going away anytime ...

File: source/_drafts/Video-js-7-Roadmap.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+When Video.js 7 is released it’ll be the end-of-life for Video.js 5.x from a support perspective. Of course, we are not going to unpublish this version from npm or remove the files from the CDN, so you are free to continue using it. However, we urge you to update at your earliest convenience as we will no longer be accepting fixes.
+
+## Google Analytics
+
+{% pullquote right %}
+![](fastly small.png)
+Thanks [Fastly](https://www.fastly.com/) for sponsoring vjs.zencdn.net
+{% endpullquote %}
+
+Video.js provides CDN hosted files for Video.js, sponsored by Fastly -- thanks Fastly! These files currently have a stripped down pixel tracking for Google Analytics (GA) that sends limited information to a GA account we own. We did a terrible job communicating that this is happening and how users can opt out -- set window.HELP_IMPROVE_VIDEOJS to false before loading in Video.js -- or use another CDN like unpkg or CDNJS. Recently we made a change that makes this tracking pixel honor the Do Not Track flag set by users in the browser, however, this isn’t going to affect previously released versions of Video.js
```
Comment (mmcc) [inline source/_drafts/Video-js-7-Roadmap.md]: Should we make the first or second Fastly inline a link as well?

File: source/_drafts/Video-js-7-Roadmap.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+![](remove GA.png)
+In addition, starting with Video.js 7.0 (and probably back ported to newer versions of 6.x) we will *no longer* be including this tracking pixel in our CDN builds. Instead, we are investigating our options for using the CDN logs via Fastly. We expect to share more details as we get closer to the Video.js 7 release date and have a better idea of what we will be doing.
```
Comment (mmcc) [inline source/_drafts/Video-js-7-Roadmap.md]: No need for passive voice:

> we will *no longer* include this tacking pixel

**Review by mmcc — APPROVED** [review]
Awesome!

File: source/_drafts/Video-js-7-Roadmap.md (hunk 4 above)
Comment (gkatsev) [reply source/_drafts/Video-js-7-Roadmap.md]: Sure

**Review by kcj382 — APPROVED** [review]
Looks great!

**Review by mmcc — APPROVED** [review]
👍 👍

**Pull request merged**